mobile/src/androidTest/java/org/openhab/habdroid/ui/BasicWidgetTest.java

```diff
@@ -45,48 +49,25 @@
 public class BasicWidgetTest {
 
     @Rule
-    public ActivityTestRule<OpenHABMainActivity> mActivityTestRule = new ActivityTestRule<>(OpenHABMainActivity.class);
+    public ActivityTestRule<OpenHABMainActivity> mActivityTestRule = new ActivityTestRule<>
+            (OpenHABMainActivity.class, true, false);
 
     private IdlingResource mProgressbarIdlingResource;
 
-    @Test
-    public void openHABMainActivityTest2() throws InterruptedException {
-        // click next
-        ViewInteraction appCompatImageButton = onView(
-                allOf(withId(R.id.next),
-                        childAtPosition(
-                                allOf(withId(R.id.bottomContainer),
-                                        childAtPosition(
-                                                withId(R.id.bottom),
-                                                1)),
-                                3),
-                        isDisplayed()));
-        appCompatImageButton.perform(click());
-
-        // click next
-        ViewInteraction appCompatImageButton2 = onView(
-                allOf(withId(R.id.next),
-                        childAtPosition(
-                                allOf(withId(R.id.bottomContainer),
-                                        childAtPosition(
-                                                withId(R.id.bottom),
-                                                1)),
-                                3),
-                        isDisplayed()));
-        appCompatImageButton2.perform(click());
+    @Before
+    public void setup() {
+        PreferenceManager
+                .getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext())
+                .edit()
```

Comment:
I tried first calling clear() here. However, even if it works perfectly fine for the first test run, during the second test run, the preferences are resetted to the default values and the ones I set next are applied, however, the app behaves weird when started:
* The theme is dark, not light, even though that light is selected in the preferences
* the discovery of the openHAB server is started, even though Demo mode is enabled in the preferences
* The app says, that I should check, if openHAB is running

Opening the settings and just going back will result in the expected behaviour (light theme, demo mode, sitemap selection). I've no idea why this happens, however it does not happen when not using clear().

This may not be important for the tests in travis, as uninstalling the app before running the test again also works (and travis should always install on a clean emulator), however, I don't think that it will help us as developers :)